src/components/replacement/bh.rs

```diff
@@ -0,0 +1,62 @@
+//! Replacement components for the Black Hole algorithm (BH).
+
+use rand::Rng;
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    population::BestIndividual, problems::LimitedVectorProblem, utils::squared_euclidean,
+    Component, ExecResult, Individual, SingleObjectiveProblem, State,
+};
+
+#[derive(Clone, Serialize, Deserialize)]
+pub struct EventHorizon;
+
+impl EventHorizon {
+    pub fn new<P>() -> Box<dyn Component<P>>
+    where
+        P: LimitedVectorProblem<Element = f64>,
+        P: SingleObjectiveProblem,
+    {
+        Box::new(Self)
+    }
+}
+
+impl<P> Component<P> for EventHorizon
+where
+    P: LimitedVectorProblem<Element = f64>,
+    P: SingleObjectiveProblem,
+{
+    fn execute(&self, problem: &P, state: &mut State<P>) -> ExecResult<()> {
+        let mut populations = state.populations_mut();
+        let offspring = populations.pop();
+
+        let f_bh = state.best_objective_value().unwrap().value();
+
+        let fitness_sum = offspring.iter().map(|x| x.objective().value()).sum::<f64>();
+        let radius = f_bh / fitness_sum;
+
+        let best_ind = offspring.best_individual().cloned();
+        let best = best_ind.unwrap().solution().clone();
+        let distances = offspring
+            .iter()
+            .map(|o| squared_euclidean(o.solution(), &best).sqrt())
+            .collect::<Vec<f64>>();
```

Comment:
If the best individual is only needed as a reference for calculating the distance to it, wouldn't 
```
let best = offspring.best_individual().unwrap();
```
and then `squared_euclidean(o.solution(), best.solution())` be enough here?